Lesson5/Task1/Program.cs

```diff
@@ -0,0 +1,110 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Task1 {
+    class RatNum {
+        private int m;
+        private int n;
+
+        private static int GCD(int a, int b) {
+            if (b < 0) b = -b;
+            if (a < 0) a = -a;
+            while (b != 0) {
+                int t = b;
+                b = a % b;
+                a = t;
+            }
+            return a;
+        }
+
+        private static int LCM(int a, int b) => Math.Abs(a * b) / GCD(a, b);
+
+        public RatNum(int _m, int _n) {
+            if (_n <= 0) throw new ArgumentException("Denominator can't be zero.");
+            m = _m;
+            n = _n;
+        }
+
+        public RatNum(int _m) {
+            m = _m;
+            n = 1;
+        }
+
+        public RatNum Reduce { 
+            get {
+                int t = GCD(m, n);
+                return new RatNum(m / t, n / t);
+            }
+        }
+
+        public static bool operator ==(RatNum a, RatNum b) => ((a.m * b.n) == (b.m * a.n));
```

Comment:
Оператор равенства обычно реализуется через вызов Equals(a, b)
```suggestion
        public static bool operator ==(RatNum a, RatNum b) => Equals(a, b);
```